app/components/UI/TransactionElement/TransactionDetails/index.js

```diff
@@ -166,6 +174,39 @@ class TransactionDetails extends PureComponent {
     return await query(ethQuery, 'getTransactionReceipt', [transactionHash]);
   };
 
+  /**
+   * Returns the appropriate block explorer URL for a given chain
+   * @param {string} chainId - The chain ID to get the block explorer for
+   * @param {string} txChainId - The transaction chain ID
+   * @param {Object} networkConfigurations - The network configurations object
+   * @returns {string} The block explorer URL
+   */
+  getBlockExplorerForChain = (chainId, txChainId, networkConfigurations) => {
+    // First check for network configuration block explorer
+    let blockExplorer =
+      networkConfigurations?.[txChainId]?.blockExplorerUrls[
+        networkConfigurations[txChainId]?.defaultBlockExplorerUrlIndex
+      ] || NO_RPC_BLOCK_EXPLORER;
+
+    // Check for default block explorers based on chain ID
+    if (isMainNet(txChainId)) {
+      blockExplorer = MAINNET_BLOCK_EXPLORER;
+    } else if (isLineaMainnetChainId(txChainId)) {
+      blockExplorer = LINEA_MAINNET_BLOCK_EXPLORER;
+    } else if (txChainId === CHAIN_IDS.LINEA_SEPOLIA) {
+      blockExplorer = LINEA_SEPOLIA_BLOCK_EXPLORER;
+    } else if (txChainId === CHAIN_IDS.SEPOLIA) {
+      blockExplorer = SEPOLIA_BLOCK_EXPLORER;
+    }
```

Comment:
```suggestion
    // Due to `blockExplorerUrls` may be omit for new installed user,
    // hence, we re-assign the default block explorer from `blockExplorerUrls` 
    // TODO: It should be fixed in NetworkController
    if (blockExplorer === NO_RPC_BLOCK_EXPLORER  && hasProperty(BlockExplorerUrl, txChainId) {
       blockExplorer = BlockExplorerUrl[txChainId]
    }
```

URL can be found from `controler-utils`